src/System.Private.CoreLib/shared/System/String.Comparison.cs

```diff
@@ -566,7 +566,7 @@ public bool EndsWith(string value, StringComparison comparisonType)
                     return CompareInfo.Invariant.IsSuffix(this, value, GetCaseCompareOfComparisonCulture(comparisonType));
 
                 case StringComparison.Ordinal:
-                    return this.Length < value.Length ? false : (CompareOrdinalHelper(this, this.Length - value.Length, value.Length, value, 0, value.Length) == 0);
+                    return this.Length < value.Length ? false : this.AsSpan(Length - value.Length).SequenceEqual(value);
```

Comment:
And this ends up being better than:
```C#
return this.Length >= value.Length && this.AsSpan(Length - value.Length).SequenceEqual(value);
```
?

Reply:
Went for second as is nice and clear, however just going to check if one of the patterns elides the check in the span .ctor

Reply:
Sorta...